Automated Smoketest: :x:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 Host: ip-10-20-0-69.ca-central-1.compute.internal
 GCC:  gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Git: 2.9.5,  CMake: 3.22.1,  cUrl: 7.67.0,  node.js: v16.13.1,  npm: 8.1.2
Sha: 9e485b76ab920c9b668230e62ca568f289354e54
Containerized:False
Number of Error(s): 7
/home/centos/smoketest/PR-15491/HPCC-Platform/plugins/nlp/nlp.cpp:46:60: error: \xe2\x80\x98parentCtx\xe2\x80\x99 was not declared in this scope
make[2]: *** [plugins/nlp/CMakeFiles/nlp.dir/build.make:76: plugins/nlp/CMakeFiles/nlp.dir/nlp.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:15096: plugins/nlp/CMakeFiles/nlp.dir/all] Error 2
warn(226) A package without a profile could throw errors or warnings. package: dojo-util
warn(226) A package without a profile could throw errors or warnings. package: dojo-util
\terrors: 0
make: *** [Makefile:166: all] Error 2
Get ECLWatch build result
ReportTimes.
Finish at: 2022-04-13_14-55-56
time logs started
Prep time   :    24 sec (00:00:24)
Build time  :   164 sec (00:02:44)
Package time:     0 sec (00:00:00)
Install time:     0 sec (00:00:00)
Start time  :     0 sec (00:00:00)
Test time   :     0 sec (00:00:00)
Stop time   :     0 sec (00:00:00)
Summary     :   188 sec (00:03:08)
time logs end

[Build error:](https://gist.github.com/5d20996e065dafb591ee3c24b0749c60)
Build log [Build log](https://gist.github.com/HPCCSmoketest/5d20996e065dafb591ee3c24b0749c60#file-result-22-04-13-14-52-31-log)

--- a/plugins/nlp/nlp.cpp
+++ b/plugins/nlp/nlp.cpp
@@ -49,8 +49,6 @@ namespace nlp {
     static CriticalSection cs; 
     static NLPEng *nlpEng = NULL;
 
-    ECL_NLP_API void setPluginContext(IPluginContext * _ctx) { parentCtx = _ctx; }
-
     //--------------------------------------------------------------------------------
     //                           ECL SERVICE ENTRYPOINTS
     //--------------------------------------------------------------------------------
@@ -71,3 +69,7 @@ namespace nlp {
         memcpy_iflen(tgt, sso.str().c_str(), tgtLen);
     }
 } // namespace nlp
+
+using namespace nlp;
+
+ECL_NLP_API void setPluginContext(IPluginContext * _ctx) { parentCtx = _ctx; }

--- a/plugins/nlp/nlp.hpp
+++ b/plugins/nlp/nlp.hpp
@@ -35,4 +35,10 @@
 #include "nlp_eng.hpp"
 #include <sstream>
 
+extern "C"
+{
+    ECL_NLP_API bool getECLPluginDefinition(ECLPluginDefinitionBlock *pb);
+    ECL_NLP_API void setPluginContext(IPluginContext * _ctx);
+}
+
 #endif

